gapis/api/vulkan/graph_visualization.go

```diff
@@ -66,60 +75,129 @@ func getSubCommandHierarchyNames() *api.HierarchyNames {
+	for i := from; i <= to; i++ {
+		builder.labelsInsideDebugMarkers[i].Insert(level, VK_CMD_DEBUG_MARKER, builder.numberOfDebugMarker)
+	}
+}
+
+func checkDebugMarkers(builder *labelForVulkanCommands, currentLabel *api.Label) {
+	commandName := currentLabel.GetCommandName()
+	positionOfLastDebugMarkerBegin := 0
+	labelOfLastDebugMarkerBegin := &api.Label{}
+	if len(builder.positionOfDebugMarkersBegin) > 0 {
+		positionOfLastDebugMarkerBegin = builder.positionOfDebugMarkersBegin[len(builder.positionOfDebugMarkersBegin)-1]
+		labelOfLastDebugMarkerBegin = builder.labelsInsideDebugMarkers[positionOfLastDebugMarkerBegin]
+	}
+
+	if commandName == VK_CMD_DEBUG_MARKER_BEGIN {
+		if len(builder.positionOfDebugMarkersBegin) > 0 {
+			if currentLabel.GetSize() < labelOfLastDebugMarkerBegin.GetSize() {
+				builder.positionOfDebugMarkersBegin = builder.positionOfDebugMarkersBegin[:len(builder.positionOfDebugMarkersBegin)-1]
+			}
+		}
+		builder.labelsInsideDebugMarkers = append(builder.labelsInsideDebugMarkers, currentLabel)
+		builder.positionOfDebugMarkersBegin = append(builder.positionOfDebugMarkersBegin, len(builder.labelsInsideDebugMarkers)-1)
+
+	} else if commandName == VK_CMD_DEBUG_MARKER_END {
+
+		if len(builder.positionOfDebugMarkersBegin) > 0 {
+			if currentLabel.GetSize() != labelOfLastDebugMarkerBegin.GetSize() {
+				builder.positionOfDebugMarkersBegin = builder.positionOfDebugMarkersBegin[:len(builder.positionOfDebugMarkersBegin)-1]
+
+			} else if getMaxCommonPrefix(labelOfLastDebugMarkerBegin, currentLabel) == currentLabel.GetSize()-1 {
+				builder.labelsInsideDebugMarkers = append(builder.labelsInsideDebugMarkers, currentLabel)
+				builder.positionOfDebugMarkersBegin = append(builder.positionOfDebugMarkersBegin, len(builder.labelsInsideDebugMarkers)-1)
+				addDebugMarker(builder, positionOfLastDebugMarkerBegin, len(builder.labelsInsideDebugMarkers)-1)
+				builder.positionOfDebugMarkersBegin = builder.positionOfDebugMarkersBegin[:len(builder.positionOfDebugMarkersBegin)-1]
+			}
+		}
+
+	} else if len(builder.positionOfDebugMarkersBegin) > 0 {
+		if currentLabel.GetSize() < labelOfLastDebugMarkerBegin.GetSize() {
+			builder.positionOfDebugMarkersBegin = builder.positionOfDebugMarkersBegin[:len(builder.positionOfDebugMarkersBegin)-1]
+		} else {
+			builder.labelsInsideDebugMarkers = append(builder.labelsInsideDebugMarkers, currentLabel)
+		}
+	}
+}
+
+func getCommandBufferId(command api.Cmd) (VkCommandBuffer, bool) {
 	parameters := command.CmdParams()
 	for _, parameter := range parameters {
 		if parameter.Name == COMMAND_BUFFER {
-			commandBuffer := fmt.Sprintf("%s_%d", parameter.Name, parameter.Get())
-			return commandBuffer
+			return parameter.Get().(VkCommandBuffer), true
 		}
 	}
-	return ""
+	return 0, false
 }
 
-func (builder *labelForVulkanCommands) GetCommandLabel(command api.Cmd, commandNodeId uint64) string {
+func (builder *labelForVulkanCommands) GetCommandLabel(command api.Cmd, cmdId uint64) *api.Label {
```

Comment:
These functions that implement the interfaces are also public, and should also have godocs. Those godocs are usually exactly the same as the godocs from the interface method they are implementing.

Reply:
Thank you, I will add the Godocs here as well.